api/api/validators/generate_teams_validator.py

```diff
@@ -49,31 +52,148 @@ def validate_algorithm_options(self):
         pass
 
     def validate_students(self):
-        # todo: validate student ids are unique
-        # todo: validate that if project preferences are specified, then the project ids exist in the team options
-
-        # used to validate a Student dataclass object
-        # Schema(int).validate(self.id)
-        # if self.name:
-        #     Schema(str).validate(self.name)
-        # Schema(dict).validate(self.relationships)
-        # for student_id, relationship in self.relationships.items():
-        #     Schema(str).validate(student_id)
-        #     Schema(Relationship).validate(relationship)
-        # Schema(dict).validate(self.attributes)
-        # Schema(list).validate(self.project_preferences)
-
-        # will be converted to look more like:
-        # Schema([
-        #     {
-        #         "_id": int,
-        #
-        #     }
-        # ]).validate(self.data)
+        students: List = self.data.get("students")
+        Schema(
+            [
+                {
+                    "id": int,
+                    Optional("name"): str,
+                    Optional("attributes"): {str: [int]},
+                    Optional("relationships"): {
+                        str: Or(
```

Comment:
same here with the string technically being an int

Reply:
line 136 checked that